account_consolidation/wizard/consolidation_base.py

```diff
@@ -0,0 +1,142 @@
+                 company account
+        """
+        subsidiary_accounts = self.env['account.account'].search([
+            ('company_id', '=', subsidiary.id)])
+
+        mapping_errors = {}
+
+        for account in subsidiary_accounts:
+            account_errors = []
+            if not account.consolidation_account_id:
+                account_errors.append(_(
+                    'No consolidation account defined for this account'))
+                mapping_errors.update({account: account_errors})
+                continue
+
+            conso_acc = account.consolidation_account_id
+
+            if conso_acc not in conso_holding_accounts:
+                if conso_acc.company_id != self.company_id:
+                    account_errors.append(_(
+                        'The consolidation account defined for this account '
+                        'should be on company %s.') % self.company_id.name)
+
+                mapping_errors.update({account: account_errors})
+
+        return mapping_errors
+
+    def check_account_mapping(self):
+        """Check subsidiaries mapping against hodling's."""
+        self.ensure_one()
+
+        invalid_items_per_company = {}
+
+        conso_holding_accounts = self.env['account.account'].search([
+            ('company_id', '=', self.company_id.id)])
+
+        for subsidiary in self.company_id.consolidation_profile_ids.mapped(
+                'sub_company_id'):
+
+            invalid_items = self.check_subsidiary_mapping(
+                conso_holding_accounts, subsidiary)
+
+            if any(invalid_items):
+                invalid_items_per_company[subsidiary] = invalid_items
+
+        return invalid_items_per_company
+
+    def check_interco_partner(self):
+        """
+        Check that subsidiaries' partners defined in profiles are not linked
+        to a company.
+        """
+        self.ensure_one()
+
+        invalid_partners = {}
+
+        conso_profiles = self.company_id.consolidation_profile_ids
+        for subsidiary in conso_profiles.mapped('sub_company_id'):
+            partner = subsidiary.partner_id
+            if partner.company_id:
+                invalid_partners[partner] = partner.company_id
+
+        return invalid_partners
+
+    def check_companies_allowed(self):
+        """
+        Check that the user has access to subsidiaries defined in profiles.
+        """
+        self.ensure_one()
+
+        subsidiaries = self.company_id.consolidation_profile_ids.mapped(
+            'sub_company_id')
+
+        return [s for s in subsidiaries if s not in self.env.user.company_ids]
+
+    @api.multi
+    def run_consolidation(self):
+        """Consolidate.
+
+        Proceed with all checks before launch any consolidation step.
+        This is a base method intended to be inherited with the next
+        consolidation steps.
+        """
+        self.ensure_one()
+
+        if (
+                self.check_account_mapping() or
```

Comment:
IMO, it's better to open a next line w/ an operator instead of leaving it hanging on a previous line